### PR TITLE
Add `organized_by` into context

### PR DESCRIPTION
**Pull request by gee-forr**
Hi there,

TL;DR - Context's now have an `#organized_by` attr.

We recently came across an issue where we were using actions directly, outside of an organizer. When they failed, and triggered a roll back (like when they're used with an organizer), they instead raise an `FailWithRollbackError` exception, which is used internally by LS for managing roll back flow.

This is fine, but there are times when we need to just call the action directly. Rather than duplicate the action, and make it fail without a rollback, I added the ability for a context to know if it's part of an organized run, or a single action.

We can then put an (admittedly ugly) if/else block and fail it differently depending on whether or not the action is called directly or from an organizer.

Hope you like, please let me know if there's anything I need to change.

e.g.

```ruby
class FooAction
  extend LightService::Action

  executed do |ctx|
    if ctx.organized_by.nil? # Running in an action, fail with no rollback
      ctx.fail!
    else # ctx.organized_by would == FooOrganizer, fail with rollback
      ctx.fail_with_rollback!
    end
  end
end
```

### CODE REVIEW

**Comment by adomokos** [comment]
This is a great PR, thanks for it @gee-forr! I'll review it later today. Thank you for it!

**Comment by adomokos** [comment]
It seems there Rubocop violations, according to Travis CI, can you please fix those?

```
3.41s$ bundle exec rubocop
Inspecting 83 files
............................................................C....C.................
Offenses:
spec/acceptance/organizer/iterate_spec.rb:24:81: C: Metrics/LineLength: Line is too long. [83/80]
    result = TestDoubles::TestIterate.call(:number => 1, :counters => [1, 2, 3, 4])
                                                                                ^^^
spec/acceptance/organizer/reduce_if_spec.rb:53:81: C: Metrics/LineLength: Line is too long. [83/80]
    result = TestDoubles::TestIterate.call(:number => 1, :counters => [1, 2, 3, 4])
                                                                                ^^^
83 files inspected, 2 offenses detected
The command "bundle exec rubocop" exited with 1.
```

**Comment by gee-forr** [comment]
You got it.

--

Gabriel Fortuna
gee.forr@gmail.com

**Comment by gee-forr** [comment]
All done :) Let me know if this will make it into a release, we'd love to use it from straight from rubygems in prod.

File: lib/light-service/context.rb [hunk 1]
```diff
@@ -8,16 +8,17 @@ module Outcomes
...
     def initialize(context = {},
                    outcome = Outcomes::SUCCESS,
                    message = '',
                    error_code = nil)
-      @outcome = outcome
-      @message = message
-      @error_code = error_code
+      @outcome        = outcome
+      @message        = message
+      @error_code     = error_code
```
Comment (adomokos) [inline lib/light-service/context.rb:19]: Is this a fly-by?

File: spec/acceptance/organizer/reduce_if_spec.rb [hunk 2]
```diff
@@ -49,6 +49,13 @@ def self.actions
     expect(result).to be_success
   end
 
+  it "knows that it's being reduced from within an organizer" do
+    result = TestDoubles::TestIterate.call(:number => 1,
+                                           :counters => [1, 2, 3, 4])
+
+    expect(result.organized_by).to eq TestDoubles::TestIterate
```
Comment (adomokos) [inline spec/acceptance/organizer/reduce_if_spec.rb:56]: Did you want to test it with `reduce_if`?

File: spec/action_spec.rb [hunk 3]
```diff
@@ -68,6 +68,14 @@
     expect(result.to_hash).to eq(:number => 2)
   end
 
+  context "when called directly" do
+    it "is expected to not be organized" do
+      result = TestDoubles::AddsTwoActionWithFetch.execute(context)
+
+      expect(result.organized_by).to be_nil
```
Comment (adomokos) [inline spec/action_spec.rb:75]: Sweet! 😍

File: lib/light-service/context.rb (hunk 1 above)
Comment (gee-forr) [reply lib/light-service/context.rb:19]: Yeah, I aligned those assignments, because they were underneath a multiline argument list, and needed to read them more easily. I must've mistakenly committed it instead of undoing it when I was finished with the code. I'll undo it :)

File: spec/acceptance/organizer/reduce_if_spec.rb (hunk 2 above)
Comment (gee-forr) [reply spec/acceptance/organizer/reduce_if_spec.rb:56]: Yes - I wanted to make sure all `reduce_if`, `reduce_until`, and `iterate` "meta-organizers" were handled. I've amended the expectation text to make their test cases a little clearer.

File: spec/action_spec.rb (hunk 3 above)
Comment (gee-forr) [reply spec/action_spec.rb:75]: Thank :)

File: spec/acceptance/organizer/reduce_if_spec.rb (hunk 2 above)
Comment (adomokos) [reply spec/acceptance/organizer/reduce_if_spec.rb:56]: I meant you used `TestIterate`, when the file itself is `reduce_if_spec`. I'll merge it, but will fix it.

**Comment by adomokos** [comment]
Thank you! 💯  👍

**Comment by adomokos** [comment]
Gem released with your changes under version [0.14.0](https://rubygems.org/gems/light-service/versions/0.14.0).
